client/src/App.tsx

```diff
@@ -6,19 +6,39 @@ import { Registration } from './pages/Registration';
 import { ProductDetails } from './pages/ProductDetails';
 import { NotFound } from './pages/NotFound';
 import { Home } from './pages/Home';
+import { useState } from 'react';
+import { closeContext } from './components/NavDrawerCloseContext';
 
 export default function App() {
+  const [isDrawerOpen, setIsDrawerOpen] = useState(false);
```

Comment:
I'm impressed with your work on making the drawer close by using context, putting the open/close functions in your context, and passing `handleClose` functions to all your components. It is work well done.

I've been thinking, however, that there might be a simpler solution, one that will also fix the "click on body" problem that you pointed out. (You probably want to delay working on this until near the end of your project, though.)

The first solution would be to use a "Shade" overlay like we did in the app-drawer exercise. That way, when the user opens the drawer, the rest of the app becomes unclickable until they make a selection. If they do click on the rest of the app, instead of taking an action, it merely closes the drawer. I personally find this UI quite natural.

If, however, you still want the rest of the app to be clickable, then another solution is needed. Here's a thought. The first observation is the _every_ time the browser URL changes (that is, the `location` changes), you want to close the drawer. If this is not the case, then what you've done is required. But I believe that is what you want.
You could make use of React Router's `useLocation` hook and `useEffect` to do this. It would look something like:
```
  const location = useLocation();

  useEffect(() => {
    close();
  }, [location])
```
This works because every time the location changes, the useEffect is executed, which closes the drawer.